lib/src/core/verifier/policy/opa/mod.rs

```diff
@@ -0,0 +1,109 @@
+use anyhow::{anyhow, Result};
+use std::ffi::CStr;
+use std::os::raw::c_char;
+
+/// Link import cgo function
+#[link(name = "opa")]
+extern "C" {
+    pub fn evaluateGo(policy: GoString, data: GoString, input: GoString) -> *mut c_char;
+}
+
+/// String structure passed into cgo
+#[derive(Debug)]
+#[repr(C)]
+pub struct GoString {
+    pub p: *const c_char,
+    pub n: isize,
+}
+
+pub fn evaluate(policy: String, reference: String, input: String) -> Result<String> {
+    let policy_go = GoString {
+        p: policy.as_ptr() as *const i8,
+        n: policy.len() as isize,
+    };
```

Comment:
consider adding a `from_str(s: String)` function in `impl GoString` with this functionality.

Reply:
I tried to add the following codes, but it will cause execution failure. It is weired.
```
fn from_str(s: String) -> GoString {
    GoString {
        p: s.as_ptr() as *const i8,
        n: s.len() as isize,
    }  
}
```